### PR TITLE
feat: setup cypress

### PR DESCRIPTION
**Pull request by glebez**
Hi, 

this PR is an attempt of first step for issue #776 

It includes cypress setup, new npm scripts and one simple test. I've left most of the cypress scaffolded files, since it can be a good reference for other contributors.

Currently, cypress tests can not be run fully automatically, so first you need to start a test server (`test:cypress:runTestServer`) and than either run the cypress GUI (`test:cypress:open`) or run the tests from command line (`test:cypress:run`).

I have reused the server from `run.server.js`, but changed the port to 8082 so the tests can be run in parallel with default port styleguidist instance.

Would love to get some feedback and maybe a little better understanding where do you see cypress tests fit for this project. Do you want them to run together with other tests on commit or run them in CI?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -102,6 +102,7 @@
     "babel-preset-env": "^1.6.1",
     "babel-preset-react": "^6.24.1",
     "cheerio": "^1.0.0-rc.2",
+    "cypress": "^1.4.1",
```
Comment (sapegin) [inline package.json]: Is it big / slow to install like Puppeteer?

File: .eslintrc [hunk 2]
```diff
@@ -3,12 +3,14 @@
   "extends": "tamia/react",
   "env": {
     "browser": true,
-    "node": true
+    "node": true,
+    "cypress/globals": true
```
Comment (sapegin) [inline .eslintrc]: I'm not sure it's really useful, I'd just define `cy` in globals.

Comment (glebez) [reply .eslintrc]: I've just seen the recommendation to add this in the cypress docs, so don't have a strong opinion here. It also adds browser and mocha globals, but those might be in your setup already.

File: package.json (hunk 1 above)
Comment (glebez) [reply package.json]: Hm, not sure, will check. You want to move it the installation to pre-test hook, like you're doing with puppeteer?

File: .eslintrc (hunk 2 above)
Comment (sapegin) [reply .eslintrc]: I don't have a strong opinion also but it feels unnecessary to have an extra dependency to basically define one or tow global variables in ESLint config.

File: package.json (hunk 1 above)
Comment (sapegin) [reply package.json]: Only if they do something crazy: Prettier downloads Chrome, which is ~70 MB, on install.

**Comment by glebez** [comment]
So, the requested changes are there and travis runs cypress successfully 🎉 
And yes, I removed cypress from dependencies, since it was installing 89 other packages and for me it took about a minute on overall installation. So now it is installed through a script.

I'm wondering if the cypress tests should run only for node 8, like current browser test, or it's ok that they run for all versions, like now?

**Comment by sapegin** [comment]
Awesome!

> I'm wondering if the cypress tests should run only for node 8, like current browser test, or it's ok that they run for all versions, like now?

I think we should only run it once per CI run, I don't see how it could be useful to run it in multiple Node versions -- unit tests should be enough to test Node versions compatibility.

**Comment by glebez** [comment]
All right, now it runs only once per CI run.

**Comment by sapegin** [comment]
Awesome, merged!

**Pull request merged**